src/nrnpython/nrnpy_hoc.cpp

```diff
@@ -2064,6 +2092,45 @@ static int hocobj_setitem(PyObject* self, Py_ssize_t i, PyObject* arg) {
     return err;
 }
 
+static int hocobj_slice_setitem(PyObject* self, PyObject *slice, PyObject* arg)
+{
+    // Non slice indexing still uses original function
+    if (!PySlice_Check(slice)) {
+        return hocobj_setitem(self, PyLong_AsLong(slice), arg);
+    }
+    auto *po = (PyHocObject *) self;
+    if (!po->ho_) {
+        PyErr_SetString(PyExc_TypeError, "Obj is NULL");
+        return -1;
+    }
+    if (po->type_ != PyHoc::HocObject || po->ho_->ctemplate != hoc_vec_template_) {
+        PyErr_SetString(PyExc_TypeError, "sequence index must be integer, not 'slice'");
+        return -1;
+    }
+    Vect* v = (Vect*) po->ho_->u.this_pointer;
+    Py_ssize_t start = 0;
+    Py_ssize_t end = 0;
+    Py_ssize_t step = 0;
+    Py_ssize_t slicelen = 0;
+    Py_ssize_t cap = vector_capacity(v);
+    PySlice_GetIndicesEx(slice, cap, &start, &end, &step, &slicelen);
+    // Slice index assignment requires a list of the same size as the slice
+    if (PyList_Check(arg)) {
+        Py_ssize_t arg_sz = PyList_Size(arg);
+        if (arg_sz != slicelen) {
+            PyErr_SetString(PyExc_IndexError, "list must be same size as slice");
+            return -1;
+        }
+    } else {
+        PyErr_SetString(PyExc_TypeError, "can only assign an iterable");
+        return -1;
+    }
+    for (ssize_t i = 0; i < slicelen; ++i) {
+        PyArg_Parse(PyList_GetItem(arg, i), "d", vector_vec(v) + (i * step + start));
```

Comment:
Pretty sure this won't work if we're not using a `List` for assignment, e.g. what about:
```python
>>> # let's assume a is a h.Vector
>>> gen = (x for x in (1, 2, 3))
>>> type(gen)
<class 'generator'>
>>> a[1:3] = gen
>>> list(a)
[531, 1, 2, 3, 6, 321, 1, 62]
```

Can you revise this to work with a generic iterable?